packages/react/src/components/SearchField/index.tsx

```diff
@@ -0,0 +1,93 @@
+import React, {
+  useState,
+  useRef,
+  ChangeEvent,
+  InputHTMLAttributes,
+  ReactNode,
+  forwardRef,
+  ForwardRefRenderFunction
+} from 'react';
+import classNames from 'classnames';
+
+import TextFieldWrapper from '../internal/TextFieldWrapper';
+import randomId from '../../utils/rndid';
+import Offscreen from '../Offscreen';
+import Icon from '../Icon';
+
+interface SearchFieldProps
+  extends Omit<InputHTMLAttributes<HTMLInputElement>, 'onChange'> {
```

Comment:
Just curious `HTMLAttributes<HTMLInputElement>` does not work? Just looks a bit redundant.

Reply:
Ya, it's necessary because we're overriding the `onChange` prop with a different type. Here's the error it produces if we don't omit it:
```
Interface 'SearchFieldProps' incorrectly extends interface 'InputHTMLAttributes<HTMLInputElement>'.
  Types of property 'onChange' are incompatible.
    Type '((value: string, e: ChangeEvent<HTMLInputElement>) => void) | undefined' is not assignable to type 'ChangeEventHandler<HTMLInputElement> | undefined'.
      Type '(value: string, e: ChangeEvent<HTMLInputElement>) => void' is not assignable to type 'ChangeEventHandler<HTMLInputElement>'.
        Target signature provides too few arguments. Expected 2 or more, but got 1.
```

Reply:
I think we want it to possibly be undefined in case it's an uncontrolled component